src/signals/IncidentMap/components/FilterPanel/FilterPanel.tsx

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (C) 2022 Gemeente Amsterdam
+import { useState, useEffect, Fragment } from 'react'
+
+import { ChevronLeft, ChevronRight } from '@amsterdam/asc-assets'
+import { Checkbox, Paragraph, Heading } from '@amsterdam/asc-ui'
+
+import { useFetch } from 'hooks'
+import configuration from 'shared/services/configuration/configuration'
+import type Categories from 'types/api/categories'
+
+import type { Filter } from '../../types'
+import { updateFilterCategory } from '../utils'
+import {
+  StyledPanelContent,
+  StyledLabel,
+  CategoryFilter,
+  Wrapper,
+  StyledButton,
+} from './styled'
+export type Props = {
+  filters: Filter[]
+  setFilters: (categories: Filter[]) => void
+  setMapMessage: React.Dispatch<React.SetStateAction<string>>
+}
+
+export const FilterPanel = ({ filters, setFilters, setMapMessage }: Props) => {
+  const [isOpen, setIsOpen] = useState<boolean>(true)
+
+  const { get, data, error } = useFetch<Categories>()
+
+  const toggleFilter = (categoryName: string) => {
+    const updated = updateFilterCategory(categoryName, filters)
+
+    setFilters(updated)
+  }
+
+  useEffect(() => {
+    if (filters.length === 0) {
+      get(configuration.CATEGORIES_ENDPOINT)
+    }
+  }, [filters, get])
+
+  useEffect(() => {
+    if (data?.results) {
+      const filters: Filter[] = data.results
+        .filter(({ is_public_accessible }) => is_public_accessible)
+        .map((category) => ({
+          name: category.name,
+          _display: category._display,
+          filterActive: true,
+          slug: category.slug,
+        }))
+
+      setFilters(filters)
+    }
+  }, [data, setFilters])
+
+  useEffect(() => {
+    if (error) {
+      setMapMessage('Er konden geen filter categorieën worden opgehaald.')
+      return
```

Comment:
why return here?

Reply:
maybe return null?

Reply:
haha actually have no idea. This was in here already, but I don't think it serves a role.

Reply:
oka:) might wanne remove it